src/x11_in_gtk2.c

```diff
@@ -204,8 +205,24 @@ forward_size_request(SuilX11Wrapper* socket,
 		if (hints.flags & PMinSize) {
 			width  = MAX(width, hints.min_width);
 			height = MAX(height, hints.min_height);
+			socket->hints.min_width = hints.min_width;
+			socket->hints.min_height = hints.min_height;
+			gtk_widget_set_size_request(GTK_WIDGET(&socket->socket),
+			                            socket->hints.min_width,
+			                            socket->hints.min_height);
+		}
+		if (hints.flags & PAspect) {
+			socket->hints.min_aspect = hints.min_aspect.x/hints.min_aspect.y;
+			socket->hints.max_aspect = hints.max_aspect.x/hints.max_aspect.y;
+
+			GtkWidget * top = gtk_widget_get_toplevel(GTK_WIDGET(socket->plug));
+			if (gtk_widget_is_toplevel(top)) {
+				gtk_window_set_geometry_hints(GTK_WINDOW(top),
```

Comment:
Likewise here, `forward_size_request` is called from the gtk size-allocate callback.  Apply the allocated size to all **child** widgets, do not change any top-level widgets. Their size was already allocated.